Subject: [PATCH] Set handle inheritance based on environment variable

Added logic to determine whether handles should be inherited by checking the environment variable "JB_GDB_LAUNCH_INFERIORS_WITH_HANDLES_INHERITING". This allows dynamic control of handle inheritance behavior based on the environment variable's value.
---
Index: gdb/nat/windows-nat.c
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/gdb/nat/windows-nat.c b/gdb/nat/windows-nat.c
index f9f6848861d..ed0c5719391 100644
--- a/gdb/nat/windows-nat.c
+++ b/gdb/nat/windows-nat.c
@@ -782,6 +782,18 @@ create_process_wrapper (FUNC *do_create_process, const CHAR *image,
 			INFO *startup_info,
 			PROCESS_INFORMATION *process_info)
 {
+	BOOL should_inherit_handles = FALSE;
+
+	const char *handles_inheriting_envvar =
+		getenv("JB_GDB_LAUNCH_INFERIORS_WITH_HANDLES_INHERITING");
+
+	if (handles_inheriting_envvar)
+	{
+		should_inherit_handles = strcmp(handles_inheriting_envvar, "1") == 0
+			|| strcmp(handles_inheriting_envvar, "true") == 0
+			|| strcmp(handles_inheriting_envvar, "yes") == 0;
+	}
+
   if (no_randomization && disable_randomization_available ())
     {
       static bool tried_and_failed;
@@ -827,7 +839,7 @@ create_process_wrapper (FUNC *do_create_process, const CHAR *image,
 	    {
 	      BOOL result = do_create_process (image, command_line,
 					       nullptr, nullptr,
-					       TRUE,
+					       should_inherit_handles,
 					       (flags
 						| EXTENDED_STARTUPINFO_PRESENT),
 					       environment,
@@ -853,7 +865,7 @@ create_process_wrapper (FUNC *do_create_process, const CHAR *image,
 			    command_line, /* command line */
 			    nullptr,	  /* Security */
 			    nullptr,	  /* thread */
-			    TRUE,	  /* inherit handles */
+			    should_inherit_handles,  /* inherit handles */
 			    flags,	  /* start flags */
 			    environment,  /* environment */
 			    cur_dir,	  /* current directory */
